### PR TITLE
Add float16 inference option for TFLite benchmark app

### PR DESCRIPTION
**Pull request by ysohma**
## Background
Native float16 inference is introduced in [Tensorflow Blog](https://blog.tensorflow.org/2023/11/half-precision-inference-doubles-on-device-inference-performance.html).
To try this on benchmark app(benchmark_model), we need set flag of XNNPackDelegateOptions,  it's not accessible.
This PR enable to access the flag for float16 inference from command line args.

## Change
- Add `--xnnpack_force_fp16` option in benchmark_model
  ```bash
  benchmark_model --graph=... \
    --use_xnnpack=true \
    --xnnpack_force_fp16=true # enable float16 inference.
  ```

### CODE REVIEW

**Comment by gbaned** [comment]
Hi @terryheo Can you please review this PR ? Thank you!

**Review by yijie-yang — APPROVED** [review]

File: tensorflow/lite/tools/benchmark/experimental/delegate_performance/android/src/main/native/latency_benchmark.cc [hunk 1]
```diff
@@ -157,6 +157,10 @@ std::vector<std::string> ParseArgumentsFromTfLiteSettings(
             absl::StrFormat("--num_threads=%d",
                             tflite_settings.xnnpack_settings()->num_threads()));
       }
+      if (tflite_settings.xnnpack_settings() &&
+          tflite_settings.xnnpack_settings()->flags() | TFLITE_XNNPACK_DELEGATE_FLAG_FORCE_FP16) {
```
Comment (yijie-yang) [inline tensorflow/lite/tools/benchmark/experimental/delegate_performance/android/src/main/native/latency_benchmark.cc:161]: needs to be changed to:
```
tflite_settings.xnnpack_settings()->flags() ==
              XNNPackFlags_TFLITE_XNNPACK_DELEGATE_FLAG_FORCE_FP16
```

Comment (ysohma) [reply tensorflow/lite/tools/benchmark/experimental/delegate_performance/android/src/main/native/latency_benchmark.cc]: Fixed at [de92054].(https://github.com/tensorflow/tensorflow/pull/62591/commits/de92054e43f01478d2680c4ad34199d2fb15ca0d)

Comment (yijie-yang) [reply tensorflow/lite/tools/benchmark/experimental/delegate_performance/android/src/main/native/latency_benchmark.cc]: Sorry what i meant is to move `if (tflite_settings.xnnpack_settings())` as the outer condition, and nest both `if (tflite_settings.xnnpack_settings()->num_threads()) {}` and `if (tflite_settings.xnnpack_settings()->flags() == XNNPackFlags_TFLITE_XNNPACK_DELEGATE_FLAG_FORCE_FP16) {}` within it.

Thanks!

Comment (ysohma) [reply tensorflow/lite/tools/benchmark/experimental/delegate_performance/android/src/main/native/latency_benchmark.cc]: Okay. Fixed [7c30388](https://github.com/tensorflow/tensorflow/pull/62591/commits/7c303884dfb707307f55d858fe5aec02e33d809a).
I hope this meets what you say.

**Review by yijie-yang — APPROVED** [review]

**Pull request merged**